src/alerts/check_signals_updates.R

```diff
@@ -0,0 +1,193 @@
+  }
+}
+
+#' Builds the signal alert text
+#'
+#' @param indicator_id Indicator ID
+#' @param df DataFrame with all the signals for the given indicator
+#'
+#' @returns String signal alert text
+slack_build_alert <- function(indicator_id, df) {
+  paste0(
+    "*",
+    indicator_id,
+    "* - ",
+    nrow(df),
+    " countries impacted - <",
+    df$campaign_url_archive[1],
+    " | See draft campaign>\n"
+  )
+}
+
+#' Takes the response from a GitHub Actions run of a single indicator
+#' and outputs a status message to be posted to Slack
+#'
+#' @param indicator_id ID of the indicator
+#'
+#' @returns String status message to be posted to Slack
+slack_build_workflow_status <- function(indicator_id) {
+  workflow_id <- paste0("monitor_", indicator_id, ".yaml")
+  base_logs_url <- "https://github.com/ocha-dap/hdx-signals/actions/runs/"
+
+  df_runs <- tryCatch({
+    httr2$request(
+      "https://api.github.com/repos/ocha-dap/hdx-signals/actions/workflows"
+    ) |>
+      httr2$req_url_path_append(
+        workflow_id,
+        "runs"
+      ) |>
+      httr2$req_auth_bearer_token(
+        token = get_env("GH_TOKEN")
+      ) |>
+      httr2$req_perform() |>
+      httr2$resp_body_string() |>
+      jsonlite$fromJSON(flatten = TRUE) |>
+      as.data.frame()
+  },
+  error = function(e) {
+    logger$log_error(e$message)
+    e$message
+  })
+
+  if (is.character(df_runs)) {
+    return(paste0(
+      ":red_circle: ",
+      ind,
+      ": Failed request for workflow status -",
+      df_runs,
+      "\n"
+    ))
+  }
+  # Get today's scheduled runs from the main branch
+  df_runs$date <- as.Date(df_runs$workflow_runs.created_at)
+  df_sel <- subset(
+    df_runs,
+    workflow_runs.event == "schedule" &
+      workflow_runs.head_branch == "main" &
+      date == Sys.Date()
+  )
+
+  if (nrow(df_sel) == 1) {
+    status <- df_sel[1, ]$workflow_runs.conclusion
+    run_id <- df_sel[1, ]$workflow_runs.id
+    if (status == "failure") {
+      run_link <- paste0(base_logs_url, run_id)
+      paste0(":red_circle: ", indicator_id, ": Failed update - <", run_link, "|Check logs> \n")
+    } else if (status == "success") {
+      paste0(":large_green_circle: ", indicator_id, ": Successful update \n")
+    }
+    # If no scheduled runs happened off of main today
+  } else if (nrow(df_sel) == 0) {
+    paste0(":heavy_minus_sign: ", indicator_id, ": No scheduled update \n")
+  } else {
+    paste0(":red_circle: ", indicator_id, ": More than one scheduled run today \n")
+  }
+}
+
+# Get the indicator ids of all workflows
+indicators_path <- ".github/workflows"
+indicators_files <- list.files(indicators_path)
+indicators_files <- indicators_files[stringr$str_starts(indicators_files, "monitor_")]
+indicators <- stringr$str_extract(indicators_files, "(?<=monitor_)[^\\.]+(?=\\.yaml)")
+
+full_status <- ""
+n_signals <- 0
+signals <- ""
+test <- Sys.getenv("TEST", unset = TRUE)
```

Comment:
Thinking on how this testing should be reflected. Currently, the alerts in the Slack channel do not say test. Maybe have them say  test somewhere. Oh, no! Maybe, if `HS_TEST` is `TRUE`, then messages are sent to `#hdx-signals-bot-testing` and otherwise `#hdx-signals-bot`. That way we can still test these messaging functions without disrupting a channel that should only be notified when critical things are necessary to be responded to.

Reply:
The other issue is if `HS_TEST` is `TRUE`, it compares against test data being available, which is always available now as we wait for contributors to comment. Not necessarily a problem, but we can't really use `HS_TEST` to test how Slack notifications work against what's actually going on in the repo.

Reply:
I like the idea of the two channels! I can implement that logic here.